source/math/FloatConvexHull2D.ooc

```diff
@@ -33,42 +35,99 @@ FloatConvexHull2D: class {
 		this _points add(box rightBottom)
 		this _points add(box rightTop)
 	}
+	free: override func {
+		this _points free()
+		super()
+	}
 	contains: func ~Point (point: FloatPoint2D) -> Bool {
 		result := true
-		for (i in 0 .. this count - 1)
-			if (This _isOnLeft(this _points[i], this _points[i + 1], point)) {
+		for (i in 0 .. this count)
+			if (result && This _isOnLeft(this _points[i], this _points[(i + 1) % this count], point))
```

Comment:
Why `result &&` and remove `break`? Looks like it will always iterate `count` even though it's not necessary.